frontend/src/tests/user-permission-tests.js

```diff
@@ -4,52 +4,148 @@ import { databaseSeeder } from "./setup";
 /**
  * Tests for the API. These are encapsulated in a function so that
  * different `apiGET` and `apiPOST` functions can be passed in. For example,
- * they may be functions that make actual requests via http or they may
- * be from the mock API.
- *
+ * they may be functions that make actual requests via http or they may be from
+ * the mock API.
  * @param {object} api
  * @param {Function} api.apiGET A function that when passed a route will return the get response
  * @param {Function} api.apiPOST A function that when passed a route and data, will return the post response
  */
 export function userPermissionsTests(api) {
     const { apiGET, apiPOST } = api;
 
-    const newUserData = {
-        utorid: "userxx",
+    let defaultUser;
+
+    const instructorOnlyUser = {
+        utorid: "user_permission_test_instructor_only_utorid",
         roles: ["instructor"],
     };
 
+    const taOnlyUser = {
+        utorid: "user_permission_test_ta_only_utorid",
+        roles: ["ta"],
+    };
+
+    const taInstructorUser = {
+        utorid: "user_permission_test_ta_and_instructor_utorid",
+        roles: ["ta", "instructor"],
+    };
+
+    /**
+     * Switch current active user to the user represented by the given user data. If given user does not exist, one will
+     * be created.
+     * @param newActiveUser the user to be set to the active user
+     * @returns {Promise<void>}
+     */
+    async function switchToUser(newActiveUser) {
+        // Set the active user to the target user
+        let resp = await apiPOST("/debug/active_user", newActiveUser);
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let resp = await apiPOST(`/debug/active_user`, defaultUser);
+        expect(resp).toHaveStatus("success");
+    }
+
     beforeAll(async () => {
         await databaseSeeder.seed(api);
+        let resp = await apiGET(`/debug/active_user`);
+        expect(resp).toHaveStatus("success");
+        defaultUser = {
+            utorid: resp.payload.utorid,
+            roles: resp.payload.roles,
+        };
+
+        // create in the database the user with only instructor role
```

Comment:
It's clear what the code is doing, so your comment doesn't need to address that. What you comment should address is *why* you're doing it.